bin/demo.py

```diff
@@ -2,221 +2,215 @@
+    install_package('marathon-lb', None, None, "conf/marathon-lb.json")
+
+def verify_marathon_lb(marathon_lb_url):
+    with stdchannel_redirected(sys.stderr, os.devnull):
+        try:
+            r = http.get(marathon_lb_url)
+            if r.status_code == 200 and r.text:
+                log("marathon-lb is up and running")
+                return True
+        except:
+            return False
 
 def strip_to_hostname(url):
     parsed_url = urlparse(url)
     return parsed_url.netloc
 
-def update_marathon_json(elb_url):
-    with open('marathon.json', 'r+') as f:
-        marathon = json.load(f)
-        marathon['labels']['HAPROXY_0_VHOST'] = elb_url
-        f.seek(0)
-        f.truncate()
-        json.dump(marathon, f, indent=4, sort_keys=True, separators=(',', ': '))
-
 def update_and_push_marathon_json(elb_url, branch):
     elb_hostname = strip_to_hostname(elb_url)
-    update_marathon_json(elb_hostname)
-    if call (['git', 'commit', '-a', '-m', 'Update marathon.json with ELB URL']) != 0:
-        log ("Failed to commit updated marathon.json.")
-    if call (['git', 'push', 'origin', branch]) != 0:
-        log_and_exit ("Failed to push updated marathon.json.")
-    log ("Updated marathon.json with ELB hostname {}.".format(elb_hostname))
-
-def create_credentials(jenkins_url, id, username, password):
-    credential = { 'credentials' : { 'scope' : 'GLOBAL', 'id' : id, 'username' : username, 'password' : password, 'description' : id, '$class' : 'com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl'} }
-    data = {'json' : json.dumps(credential) }
-    headers = auth_func({})
-    post_url = "{}/credential-store/domain/_/createCredentials".format(jenkins_url)
-    r = requests.post(post_url, headers=headers, data=data, verify=False)
-
-def create_job(jenkins_url, job_name, job_config):
-    log ("Creating job")
-    headers = auth_func({'Content-Type' : 'application/xml' })
-    post_url = "{}/createItem?name={}".format(jenkins_url, job_name)
-    r = requests.post(post_url, headers=headers, data=job_config, verify=False)
-    if r.status_code != 200:
-        log ("Failed to create job {} at {}.".format(job_name, jenkins_url))
-        r.raise_for_status()
-    log ("Job {} created successfully".format(job_name))
-
-def create_view(jenkins_url, view_name, view_config):
-    log ("Creating view")
-    headers = auth_func({'Content-Type' : 'text/xml' })
-    post_url = "{}/createView?name={}".format(jenkins_url, view_name)
-    r = requests.post(post_url, headers=headers, data=view_config, verify=False)
+    with open("conf/cd-demo-app.json") as options_file:
+        app_config = options_file.read().replace("ELB_HOSTNAME", elb_hostname)
+    with open("marathon.json", "w") as options_file:
+        options_file.write(app_config)
+    if call(['git', 'commit', '-a', '-m', 'Update marathon.json with ELB hostname']) != 0:
+        log_and_exit("!! failed to commit updated marathon.json")
```

Comment:
A `log` might be sufficient here: If users attempt to run this script multiple times on the same cluster, this marathon.json will not change and the commit will fail (which'll exit the demo):
```
➜  cd-demo git:(sunil-demo2) python3 bin/demo.py pipeline --password=$PASSWORD --branch=sunil-demo2 http://sunil-f4j-publicsl-9adnwfg46s3s-1406902471.us-west-2.elb.amazonaws.com/ http://sunil-f4j-elasticl-1hx6ny1cqwidv-1482094555.us-west-2.elb.amazonaws.com/
[demo]: couldn't find marathon-lb running at 'http://sunil-f4j-publicsl-9adnwfg46s3s-1406902471.us-west-2.elb.amazonaws.com/'
[demo]: installing marathon-lb

》installing package 'marathon-lb'

We recommend a minimum of 0.5 CPUs and 256 MB of RAM available for the Marathon-LB DCOS Service.
Marathon-lb DCOS Service has been successfully installed!
See https://github.com/mesosphere/marathon-lb for documentation.
On branch sunil-demo2
nothing to commit, working directory clean
[demo]: !! failed to commit updated marathon.json
```